devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+			}
+			tileAttribute("device.coolingSetpoint", key: "COOLING_SETPOINT") {
+				attributeState("coolingSetpoint", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+		}
+
+		valueTile("temperature", "device.temperature", width: 2, height: 2) {
+			state("temperature", label:'${currentValue}', unit:"dF",
+				backgroundColors:[
+							// Celsius
+							[value: 0, color: "#153591"],
+							[value: 7, color: "#1e9cbb"],
+							[value: 15, color: "#90d2a7"],
+							[value: 23, color: "#44b621"],
+							[value: 28, color: "#f1d801"],
+							[value: 35, color: "#d04e00"],
+							[value: 37, color: "#bc2323"],
+							// Fahrenheit
+							[value: 40, color: "#153591"],
+							[value: 44, color: "#1e9cbb"],
+							[value: 59, color: "#90d2a7"],
+							[value: 74, color: "#44b621"],
+							[value: 84, color: "#f1d801"],
+							[value: 95, color: "#d04e00"],
+							[value: 96, color: "#bc2323"]
+				]
+			)
+		}
+		standardTile("tempDown", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "tempDown", label:'down', action:"tempDown", defaultState: true , icon:"st.thermostat.thermostat-down"
+		}
+		standardTile("tempUp", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "tempUp", label:'up', action:"tempUp", defaultState: true , icon:"st.thermostat.thermostat-up"
+		}
+
+		valueTile("heatingSetpoint", "device.heatingSetpoint", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "heat", label:'${currentValue} heat', unit: "F", backgroundColor:"#ffffff"
+		}
+		standardTile("heatDown", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "heatDown", label:'down', action:"heatDown", defaultState: true , icon:"st.thermostat.thermostat-down"
+		}
+		standardTile("heatUp", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "heatUp", label:'up', action:"heatUp", defaultState: true , icon:"st.thermostat.thermostat-up"
+		}
+
+		valueTile("coolingSetpoint", "device.coolingSetpoint", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "cool", label:'${currentValue} cool', unit:"F", backgroundColor:"#ffffff"
+		}
+		standardTile("coolDown", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "coolDown", label:'down', action:"coolDown", defaultState: true , icon:"st.thermostat.thermostat-down"
+		}
+		standardTile("coolUp", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "coolUp", label:'up', action:"coolUp", defaultState: true , icon:"st.thermostat.thermostat-up"
+		}
+
+		standardTile("mode", "device.thermostatMode", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "updating", label:'wait', action:"", backgroundColor:"#880000", nextState:"off" //, icon: "st.secondary.secondary"
```

Comment:
Consider changing label to "Updating..." which was the recommendation our UX team gave us implementing other DTH's
Also, as action is not defined nextState is moot.
The first state is also the default state in case mode does not correspond to any of the defined state, for this reason its preferable IMHO to have the "off" state first

Reply:
Yes! changed to "Updating...".
Regarding the "off" as first state: indicating "off" while in an unknown state, may be potentially misleading to the user.  Also, in some situations the mode on UI seemed to transition through off before arriving at the correct next state.  Having updating... as first state seems more friendly.